src/hotspot/cpu/aarch64/nativeInst_aarch64.cpp

```diff
@@ -526,23 +526,20 @@ void NativeCallTrampolineStub::set_destination(address new_destination) {
 
 // Generate a trampoline for a branch to dest.  If there's no need for a
 // trampoline, simply patch the call directly to dest.
-address NativeCall::trampoline_jump(CodeBuffer &cbuf, address dest) {
+void NativeCall::trampoline_jump(CodeBuffer &cbuf, address dest) {
   MacroAssembler a(&cbuf);
-  address stub = NULL;
 
   if (a.far_branches()
       && ! is_NativeCallTrampolineStub_at(instruction_address() + displacement())) {
-    stub = a.emit_trampoline_stub(instruction_address() - cbuf.insts()->start(), dest);
+    bool result = a.emit_trampoline_stub(instruction_address() - cbuf.insts()->start(), dest);
+    assert(result, "failed to generate a trampoline stub");
+    return;
   }
 
-  if (stub == NULL) {
-    // If we generated no stub, patch this call directly to dest.
-    // This will happen if we don't need far branches or if there
-    // already was a trampoline.
-    set_destination(dest);
-  }
-
-  return stub;
+  // If we generated no stub, patch this call directly to dest.
+  // This will happen if we don't need far branches or if there
+  // already was a trampoline.
```

Comment:
I think this comment should be changed since you changed the code.

Reply:
The original code looks strange to me.
If we need a stub but we have failed to create it, we set dest anyway. This will lead to an incorrect instruction.
The behaviour is undefined. We might fail executing the instruction or jump somewhere.

Reply:
I've restored the original functionality.
Maybe Andrew(@theRealAph) can explain what should be here.